### PR TITLE
Errors on CLI server reload reconnection

### PR DESCRIPTION
**Pull request by Gonthim**
https://issues.jboss.org/browse/WFCORE-783

Also fixed a problem with the timeout argument not overriding the default or file config timeout value.

### CODE REVIEW

**Comment by bstansberry** [comment]
WildFly Core failures look related.

Please ping Alexey to have a look once the failures are sorted. This looks ok to me but he's more familiar with the existing remote reconnect logic and may notice something I don't.

File: cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java [hunk 1]
```diff
@@ -220,17 +211,8 @@ private void doHandleEmbedded(CommandContext ctx, ModelControllerClient client)
                     ctx.disconnectController();
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java]: A nit, but the comment at L203 above here is slightly out of date now. The catch of IOException is no longer just to comply with the compiler now that the call may go remote.

**Comment by Gonthim** [comment]
The server can reload into 'running' or 'restart-required' states. The persistent restart-required test wasn't running against the embedded server reload so it wasn't caught before.

File: cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java [hunk 2]
```diff
@@ -190,6 +177,10 @@ private void doHandleEmbedded(CommandContext ctx, ModelControllerClient client)
             throw new CommandLineException("Failed to execute :reload", e);
         }
 
+        ensureServerRebootComplete(ctx, client);
+    }
+
+    private void ensureServerRebootComplete(CommandContext ctx, ModelControllerClient client) throws CommandLineException {
```
Comment (aloubyansky) [inline cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java]: Note, the same "making sure it's up" issue affects ShutdownHandler when it's handling a restart.

**Comment by aloubyansky** [comment]
Looks ok to me.

**Comment by Gonthim** [comment]
Retest this please

**Comment by Gonthim** [comment]
ConnectionFactoryManagementTestCase failure seems unrelated. Can replicate locally with and without this commit. Unknown cause.

**Pull request merged**